src/api/route-services/samples.js

```diff
@@ -9,37 +11,39 @@ class SamplesService {
     this.tableName = `samples-${config.clusterEnv}`;
   }
 
-  async getSamples(experimentId) {
-    const key = convertToDynamoDbRecord({
-      experimentId,
+  async getSamples(projectUuid) {
+    const marshalledData = convertToDynamoDbRecord({
+      ':projectUuid': projectUuid,
     });
 
     const params = {
       TableName: this.tableName,
-      Key: key,
+      IndexName: 'gsiExperimentid',
+      KeyConditionExpression: 'projectUuid = :projectUuid',
+      ExpressionAttributeValues: marshalledData,
       ProjectionExpression: 'samples',
     };
     const dynamodb = createDynamoDbInstance();
 
-    const response = await dynamodb.getItem(params).promise();
+    const response = await dynamodb.query(params).promise();
 
     if (response.Item) {
```

Comment:
Is there a chance that `response` will be null/none? Because if that happens, line 30 will crash.

Reply:
Hmm... I don't think so. It should return an error `503` if there's no connection. This error should bubble up through the Express middleware